### PR TITLE
Feature/205 sap hana sql dialect

### PR DESCRIPTION
**Pull request by AnastasiiaSergienko**

### CODE REVIEW

File: jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/dialects/saphana/SapHanaSqlDialect.java [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+import java.util.*;
+
+import com.exasol.adapter.*;
+import com.exasol.adapter.capabilities.*;
+import com.exasol.adapter.dialects.*;
+import com.exasol.adapter.jdbc.*;
+import com.exasol.adapter.sql.*;
+
+/**
+ * This class implement the SAP HANA SQL dialect.
```
Comment (redcatbear) [inline jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/dialects/saphana/SapHanaSqlDialect.java]:
```suggestion
 * This class implements the SAP HANA SQL dialect.
```

File: jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/dialects/saphana/SapHanaSqlDialect.java [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+ *
+ * @see <a href=
+ *      "https://help.sap.com/viewer/4fe29514fd584807ac9f2a04f6754767/2.0.03/en-US/f327b70cae564c53a766367a8aad0164.html">SAP
```
Comment (redcatbear) [inline jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/dialects/saphana/SapHanaSqlDialect.java]: This link is portal-user-specific. Please replace by either a publicly viewable link or an abstract description of how to view the HANA SQL reference.

File: jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/dialects/saphana/SapHanaSqlDialect.java [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+public class SapHanaSqlDialect extends AbstractSqlDialect {
+    private static final String NAME = "SAPHANA";
+    private static final Capabilities CAPABILITIES = createCapabilityList();
+    private static final List<String> SUPPORTED_PROPERTIES = Arrays.asList(SQL_DIALECT_PROPERTY,
+            CONNECTION_NAME_PROPERTY, CONNECTION_STRING_PROPERTY, USERNAME_PROPERTY, PASSWORD_PROPERTY,
+            CATALOG_NAME_PROPERTY, SCHEMA_NAME_PROPERTY, TABLE_FILTER_PROPERTY, EXCLUDED_CAPABILITIES_PROPERTY,
+            DEBUG_ADDRESS_PROPERTY, LOG_LEVEL_PROPERTY);
+
+    /**
+     * Get the Sap Hana dialect name
```
Comment (redcatbear) [inline jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/dialects/saphana/SapHanaSqlDialect.java]:
```suggestion
     * Get the SAP HANA dialect name
```

Please uniformly use the original letter capitalization of the company and product name in comments.

File: jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/dialects/saphana/SapHanaSqlDialect.java (hunk 1 above)
Comment (AnastasiiaSergienko) [reply jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/dialects/saphana/SapHanaSqlDialect.java]: fixed

File: jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/dialects/saphana/SapHanaSqlDialect.java (hunk 2 above)
Comment (AnastasiiaSergienko) [reply jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/dialects/saphana/SapHanaSqlDialect.java]: replace the link. Please, try it out

File: jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/dialects/saphana/SapHanaSqlDialect.java (hunk 3 above)
Comment (AnastasiiaSergienko) [reply jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/dialects/saphana/SapHanaSqlDialect.java]: fixed

File: jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/dialects/saphana/SapHanaColumnMetadataReader.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    public SapHanaColumnMetadataReader(final Connection connection, final AdapterProperties properties,
+            final IdentifierConverter identifierConverter) {
+        super(connection, properties, identifierConverter);
+    }
+
+    @Override
+    public DataType mapJdbcType(final JdbcTypeDescription jdbcTypeDescription) {
+        if (jdbcTypeDescription.getJdbcType() == Types.DECIMAL) {
+            return DataType.createDouble();
+        }
```
Comment (redcatbear) [inline jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/dialects/saphana/SapHanaColumnMetadataReader.java]: Please use an `else` path.

File: jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/dialects/saphana/SapHanaColumnMetadataReader.java [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+     * @param identifierConverter converter between source and Exasol identifiers
+     */
+    public SapHanaColumnMetadataReader(final Connection connection, final AdapterProperties properties,
+            final IdentifierConverter identifierConverter) {
+        super(connection, properties, identifierConverter);
+    }
+
+    @Override
+    public DataType mapJdbcType(final JdbcTypeDescription jdbcTypeDescription) {
+        if (jdbcTypeDescription.getJdbcType() == Types.DECIMAL) {
```
Comment (redcatbear) [inline jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/dialects/saphana/SapHanaColumnMetadataReader.java]: Make a distinction for `DECIMALS` that have scale and precision and all others. Distinguish `DECIMAL` and `SMALLDECIMAL` by using the original name stored in `jdbcTypeDescription`.

File: jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/dialects/saphana/SapHanaColumnMetadataReader.java (hunk 4 above)
Comment (AnastasiiaSergienko) [reply jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/dialects/saphana/SapHanaColumnMetadataReader.java]: done

File: jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/dialects/saphana/SapHanaColumnMetadataReader.java (hunk 5 above)
Comment (AnastasiiaSergienko) [reply jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/dialects/saphana/SapHanaColumnMetadataReader.java]: done

File: doc/dialects/saphana.md [hunk 6]
```diff
@@ -0,0 +1,94 @@
+# SAP HANA SQL Dialect
+
+The SAP HANA SQL dialect allows you to reach [HANA](https://www.sap.com/products/hana.html) databases via Virtual Schemas.
```
Comment (redcatbear) [inline doc/dialects/saphana.md]:
```suggestion
The SAP HANA SQL dialect allows you to access [HANA](https://www.sap.com/products/hana.html) databases via Virtual Schemas.
```

File: doc/dialects/saphana.md [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+
+## JDBC Driver
+
+Download the latest version of [SAP HANA JDBC driver](https://search.maven.org/search?q=g:com.sap.cloud.db.jdbc%20AND%20a:ngdbc&core=gav).
```
Comment (redcatbear) [inline doc/dialects/saphana.md]:
```suggestion
Download the latest version of the [SAP HANA JDBC driver](https://search.maven.org/search?q=g:com.sap.cloud.db.jdbc%20AND%20a:ngdbc&core=gav).
```

File: doc/dialects/saphana.md [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+    /
+    ;
+    ```
+
+3. Create a connection
+    
+    ```sql
+    CREATE OR REPLACE CONNECTION SAPHANA_CONNECTION 
+    TO 'jdbc:sap://write.ip.address.here:port' 
+    USER 'SYSTEM'
```
Comment (redcatbear) [inline doc/dialects/saphana.md]: Don't use administrative accounts in examples.

File: doc/dialects/saphana.md [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+
+3. Create a connection
+    
+    ```sql
+    CREATE OR REPLACE CONNECTION SAPHANA_CONNECTION 
+    TO 'jdbc:sap://write.ip.address.here:port' 
+    USER 'SYSTEM' 
+    IDENTIFIED BY 'yourpassword';
+    ```
+    SAP HANA's default username is SYSTEM.
```
Comment (redcatbear) [inline doc/dialects/saphana.md]: This is an administrative account. Remove the sentence.

File: doc/dialects/saphana.md [hunk 10]
```diff
@@ -0,0 +1,94 @@
...
+
+4. Create a Virtual Schema
+
+    ```sql
+    CREATE VIRTUAL SCHEMA "saphanatest"
```
Comment (redcatbear) [inline doc/dialects/saphana.md]:
```suggestion
    CREATE VIRTUAL SCHEMA SAPHANATEST
```

Use capitalization and quoting uniformly.

Comment (AnastasiiaSergienko) [reply doc/dialects/saphana.md]: fixed

File: doc/dialects/saphana.md (hunk 9 above)
Comment (AnastasiiaSergienko) [reply doc/dialects/saphana.md]: fixed

File: doc/dialects/saphana.md (hunk 8 above)
Comment (AnastasiiaSergienko) [reply doc/dialects/saphana.md]: fixed

File: doc/dialects/saphana.md (hunk 7 above)
Comment (AnastasiiaSergienko) [reply doc/dialects/saphana.md]: fixed

File: doc/dialects/saphana.md (hunk 6 above)
Comment (AnastasiiaSergienko) [reply doc/dialects/saphana.md]: fixed

**Pull request merged**